server/app/src/main/java/com/google/heartrate/wearos/app/bluetooth/BluetoothUtils.java

```diff
@@ -0,0 +1,120 @@
+
+    /**
+     * Assert that bluetooth is supported in current device.
+     *
+     * @param context application context
+     * @throws GattException if bluetooth is not supported
+     */
+    public static void assertBluetoothIsSupported(Context context) throws GattException {
+        BluetoothManager bluetoothManager = (BluetoothManager) context.getSystemService(BLUETOOTH_SERVICE);
+
+        if (bluetoothManager == null) {
+            throw new GattException("Bluetooth is not supported. No BluetoothManager found.");
+        }
+
+        BluetoothAdapter bluetoothAdapter = bluetoothManager.getAdapter();
+
+        if (bluetoothAdapter == null) {
+            throw new GattException("Bluetooth is not supported. No BluetoothAdapter found.");
+        }
+
+        if (!bluetoothAdapter.isEnabled()) {
+            throw new GattException("Bluetooth adapter is currently disabled.");
+        }
+
+        if (!context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)) {
+            throw new GattException("Bluetooth LE  is not supported. No FEATURE_BLUETOOTH_LE found.");
+        }
+    }
+
+    /**
+     * Get bluetooth manager.
+     *
+     * @param context application context
+     * @return bluetooth manager
+     * @throws GattException if bluetooth is not supported
+     */
+    public static BluetoothManager getBluetoothManager(Context context) throws GattException {
+        BluetoothManager bluetoothManager = (BluetoothManager) context.getSystemService(Context.BLUETOOTH_SERVICE);
+
+        if (bluetoothManager == null) {
+            throw new GattException("Bluetooth is not supported. No BluetoothManager found.");
+        }
+
+        return bluetoothManager;
+    }
+
+    /**
+     * Get bluetooth adapter.
+     *
+     * @param context application context
+     * @return bluetooth adapter
+     * @throws GattException if bluetooth is not supported
+     */
+    public static BluetoothAdapter getBluetoothAdapter(Context context) throws GattException {
+        BluetoothManager bluetoothManager = getBluetoothManager(context);
+        BluetoothAdapter adapter = bluetoothManager.getAdapter();
+
+        if (adapter == null) {
+            throw new GattException("Bluetooth is not supported. No BluetoothAdapter found.");
+        }
+
+        return adapter;
+    }
+
+    /**
+     * Get bluetooth advertiser.
+     *
+     * @param context application context
+     * @return bluetooth advertiser
+     * @throws GattException if bluetooth advertising is not supported
+     */
+    public static BluetoothLeAdvertiser getBluetoothLeAdvertiser(Context context) throws GattException {
+        BluetoothAdapter adapter = getBluetoothAdapter(context);
+        BluetoothLeAdvertiser advertiser = adapter.getBluetoothLeAdvertiser();
+
+        if (advertiser == null) {
+            throw new GattException("Bluetooth advertising is not supported. Can not get BluetoothLeAdvertiser.");
```

Comment:
typo: 'Cannot'

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/4/commits/46e4bfbf8964c873edb648e1d78bd22b61070778)